.github/workflows/snap.yaml

```diff
@@ -0,0 +1,120 @@
+name: Build OpenSearch from source and run tests
+
+env:
+  VERSION: 2.2.1
+  RELEASE: edge
+
+on:
+  push:
+    branches:
+      - edge
```

Comment:
Would it not be better to publish the snap from a PR?
For this to work on a non hosted GHA runner, we would need to have the team push their branches to "canonical" (origin).

You can retrofit https://github.com/snapcore/snapcraft/blob/main/.github/workflows/publish.yaml to your needs